File: core/src/main/java/heroes/journey/entities/actions/inputs/TargetInput.java
package heroes.journey.entities.actions.inputs;

import heroes.journey.GameState;
import lombok.Getter;

import java.util.UUID;

@Getter
public class TargetInput<I> extends ActionInput {
    I input;

    public TargetInput(GameState gameState, UUID entityId, I input) {
        super(gameState, entityId);
        this.input = input;
    }
}

File: core/src/main/java/heroes/journey/entities/actions/inputs/TargetInput.kt
package heroes.journey.entities.actions.inputs

import heroes.journey.GameState
import java.util.*

class TargetInput<I>(gameState: GameState, entityId: UUID?, val input: I) : ActionInput(gameState, entityId) {
    override fun toString(): String {
        return input.toString()
    }
}
